io/disk/mvcli_test.py

```diff
@@ -0,0 +1,267 @@
+                self.log.info("Mvcli read/write mailbox register %s with \
+                              value 0 is passed", reg)
+            else:
+                self.fail("Mvcli read/write mailbox register %s with \
+                          value 0 is failed", reg)
+            self.run_command("%s 1" % write_cmd)
+            output = self.run_cmd_output(read_cmd)
+            if "register value is 0x1." in output:
+                self.log.info("Mvcli read/write mailbox register %s with \
+                              value 1 is passed", reg)
+            else:
+                self.fail("Mvcli read/write mailbox register %s with \
+                          value 1 is failed", reg)
+
+    def test_delete(self):
+        """
+        Delete virtual disk or spare drive.
+        """
+        self.log.info("Mvcli delete VD test")
+        output = self.run_cmd_output("%s info -o vd" % self.base)
+        if "No virtual disk is found." in output:
+            self.log.info("No VD's are created yet/ skipping delete \
+                          VD operation")
+            return
+        vd_list = []
+        search_string = r"id:                  (\d)"
+        for line in output.split("\n"):
+            obj = re.search(search_string, line)
+            if obj:
+                vd_list.append(obj.group(1))
+        if vd_list is None:
+            self.log.info("Not able to found any VD's are found")
+        self.log.info("List of VD's to be deleted: %s", vd_list)
+        for vd in vd_list:
+            self.run_command("%s delete -o vd -i %s" % (self.base_force, vd))
+        output = self.run_cmd_output("%s info -o vd" % self.base)
+        if "No virtual disk is found." in output:
+            self.log.info("Mvcli delete operation success")
+        else:
+            self.fail("Mvcli delete operation failed")
+
+    def test_create(self):
+        """
+        Create virtual disk on supported raid levels
+        and try to delete VD which was created
+        for RAID 0, Use all disks from PD,
+        for RAID 1, Use minimum of two disks from PD
+        for RAID 10, Use minimum of four disks from PD
+        """
+        raid_levels = ["0", "1", "10"]
+        base = "%s create -o vd " % self.base_force
+        pd_list_len = len(self.pd_ids)
+        self.log.info("PD List %s, length: %s", self.pd_ids, pd_list_len)
+        for level in raid_levels:
+            name = "MyRaid_%s" % level
+            if level == "0":
+                pd_list = ",".join(self.pd_ids)
+            elif level == "1":
+                if pd_list_len < 2:
+                    continue
+                pd_list = "%s,%s" % (self.pd_ids[0], self.pd_ids[1])
+            elif level == "10":
+                if pd_list_len < 4:
+                    continue
+                pd_list = "%s,%s,%s,%s" % (self.pd_ids[0], self.pd_ids[1],
+                                           self.pd_ids[2], self.pd_ids[3])
+            else:
```

Comment:
Based on the RAID level, list of pd's needed will vary. So list of pd's to be selected is needed based on RAID level